src/System.Net.HttpListener/tests/GetContextHelper.cs

```diff
@@ -22,30 +22,23 @@ public GetContextHelper(HttpListener listener, string listeningUrl)
             _client = new HttpClient();
         }
 
-        public async Task<HttpListenerResponse> GetResponse()
+        public HttpListenerResponse GetResponse()
         {
             // We need to create a mock request to give the HttpListener a context.
             Task<string> clientTask = _client.GetStringAsync(_listeningUrl);
```

Comment:
I hadn't noticed this is how we were achieving this.  This is unfortunate.  It means that we've still got asynchronous work running potentially after the test finishes.  But given how these tests are now structured, I don't have a good, easy suggestion, other than potentially storing the task in a field and synchronously waiting on it when disposing of the test class (ignoring exceptions, as many of these will end in failure).